### ISSUE TITLE
False positive: E1307 Argument 'builtins.NoneType' does not match format type 'd'

### ISSUE THREAD

**Issue by tardyp** [PyCQA/pylint#2631]
### Steps to reproduce

Lint following snippet on python3:
```python
# pylint: disable=C
a = None
if a is not None:
    print("%d" % (a,))

```
Real life usecase is this code:

https://github.com/buildbot/buildbot/blob/master/master/buildbot/schedulers/forcesched.py#L172
https://github.com/buildbot/buildbot/blob/master/master/buildbot/steps/python_twisted.py#L444

### Current behavior

pylint output wrong error:
```
testpylint.py:5:10: E1307: Argument 'builtins.NoneType' does not match format type 'd' (bad-string-format-type)
buildbot/schedulers/forcesched.py:172 Argument 'builtins.NoneType' does not match format type 'd' [bad-string-format-type]
buildbot/steps/python_twisted.py:444 Argument 'builtins.NoneType' does not match format type 'd' [bad-string-format-type]

```
### Expected behavior

no error

### pylint --version output

```pylint 2.2.2
astroid 2.1.0
Python 3.5.2 (default, Nov 12 2018, 13:43:14)
[GCC 5.4.0 20160609]
```

This is a regression from pylint 2.1.1

**Comment by tardyp** [comment]
looks like E1307 was added in 2.2.0 in ba62048e04fbe0e691cfccaf173c4ccbccca2992

**Comment by tardyp** [comment]
@yugr do you think it is feasible to add a check for None inspection in your checker

**Comment by yugr** [comment]
Hm, I think the real problem here is not the checking of `None` per se (it's indeed incorrect to `print("%d" % None)`) but rather that `utils.safe_infer` returns `NoneType` for `a`. Is there some general approach to work around such situations?

**Comment by PCManticore** [comment]
This is a problem with our control flow inference, which cannot understand the `if` guard clause. We have a lot of these false positives for other checks as well, all stemming from the fact that our inference engine doesn't understand flow.

**Comment by mbyrnepr2** [comment]
This one no longer reproduces (pylint 2.17.0-dev0 and astroid 2.15.0-dev0)  & should be fixed according to https://github.com/PyCQA/astroid/pull/1189.
